datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -352,6 +462,49 @@ fn build_recover_project_plan(schema: &DFSchema, input: LogicalPlan) -> LogicalP
     )
 }
 
+/// Which type of [expressions](Expr) should be considered for rewriting?
```

Comment:
While this resulted in more code, I've decided to pull the masking out into its own enum / config value so the code is easier to read and extend (I could imagine that we may wanna have more special treatments for other logical plan nodes at some point).

Reply:
Another enum also adds an excellent place for more documentation 👍